js/models/site.es6.js

```diff
@@ -27,6 +36,35 @@ Site.prototype = $.extend({},
           if(tab){
             this.trackerCount = tab.dispTotal;
           }
+      },
+
+      setHttpsMessage: function() {
+          let tab = backgroundPage.tabs[this.tabId];
+          let hostname = "www." + backgroundPage.utils.extractHostFromURL(tab.url);
+          let httpsRules = backgroundPage.all_rules.potentiallyApplicableRulesets(hostname);
+
+          if(httpsRules.size){
+              httpsRules.forEach((ruleSet) => {
+                  if(ruleSet.active && this._hasMainFrameHttpsRule(ruleSet.rules)){
+                      this.httpsState = 'default'; // figure out if this is upgraded later
+                  }
+              });
+          }
+          else if(/^https/.exec(tab.url)){
+              this.httpsState = 'default';
```

Comment:
why not like it's done here https://github.com/EFForg/https-everywhere/blob/master/chromium/rules.js#L402

using ruleset.apply to see if the url is changed?